axis_test.go

```diff
@@ -7,42 +7,57 @@ package plot
 import (
 	"math"
 	"reflect"
+	"strconv"
 	"testing"
 )
 
 func TestAxisSmallTick(t *testing.T) {
 	d := DefaultTicks{}
 	for _, test := range []struct {
 		min, max float64
+		format   func(v float64, prec int) string
```

Comment:
There should be a test where `format` is `nil`.

Reply:
This is where my lack of knowledge of the codebase comes in. Initially I added this check:

````
       if format == nil {
               format = formatFloatTick
       }
````

In `DefaultTicks.Ticks` and `LogTicks.Ticks`. However, @sbinet told me that the check wasn't required if I applied the DefaultTickFormat in `makeAxis` ( https://github.com/gonum/plot/pull/362#discussion_r136914101 ).
Right now, if you call `DefaultTicks.Ticks` directly without a format you get a panic in the test. I am assuming this will never happen but I don't have enough insight into it.

@sbinet and @kortschak please let me know what course to take on this one.